src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation getFromChar(String code) {
+            for (Operation op : Operation.values()) {
+                if (op.getCode().equals(code)) {
+                    return op;
+                }
+            }
+            throw new IllegalArgumentException("Unknown code: " + code);

```

Comment:
The `getFromChar` method name might be a bit misleading since it actually takes a `String` as an argument, not a `char`. Consider renaming it to `getFromCode` to better reflect the expected input.